docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -1,15 +1,158 @@
+  constructor(questionSlug: string, upVote: boolean, shopId: string) {
+    super(
+      QuestionVotedEventName,
+      new Date(),
+      // This are data that we want to gather
+      // The question
+      // If the question has been upvoted or downvoted
+      { questionSlug, upVote },
+      // This are the event metadata, they are relative to the event
+      // but don't contain event data, here we collect the shopId
+      // to determine on which shop the question has been voted
+      { shopId }
+    );
+  }
+}
+```
+
+:::info
+
+You can checkout the
+[ServerEvent Reference](/docs/3.x/api-reference/front-commerce-core/ServerEvent)
+to learn more about the usage of ServerEvent class.
 
 :::
+
+That's it!
+
+## Use the Event
+
+Now that we've created a Domain event class, let's use it in a route action:
+
+```tsx title="app/routes/_main.faq.$slug.tsx"
+...
+export const action = async ({
+  context,
+  params,
+  request,
+}: ActionFunctionArgs) => {
+  const { slug } = params;
+
+  const formData = await request.formData();
+  // The `action` uses the `FrontCommerceApp` class to access the Front-Commerce
+  // core from Remix. In this example, we use it to mutate data in the GraphQL
+  // unified API (similar to the one you're used to).
+  const app = new FrontCommerceApp(context.frontCommerce);
+  const response = await app.graphql.mutate(VoteIsFaqUsefulDocument, {
+    input: {
+      slug,
+      isFaqUseful: formData.get("isFaqUseful") === "true",
+    },
+  });
+
+  // highlight-start
+  // We call the `ServerEventBus` to dispatch an event.
+  // In this example, we use it to dispatch a `QuestionVotedEvent` event.
+  // We will instanciate a `QuestionVotedEvent` class with the data we want to
+  // collect for this event.
+  app.services.ServerEventBus.publish(
+    new QuestionVotedEvent(
+      String(slug),
+      formData.get("isFaqUseful") === "true",
+      app.config.currentShopId
+    )
+  );
+  // highlight-end
+
+  if (!response.voteIsFaqUseful) {
+    throw new Response("Question not found", { status: 404 });
+  }
+
+  return json({
+    success: response.voteIsFaqUseful?.success ?? false,
+  });
+};
+...
+```
+
+## Test our emitter
+
+Now that everything is ready, let's test it! To do so, start the Front-Commerce
+Worker by running:
+
+```shell
+pnpm run worker
+```
+
+The worker will boot:
+
+```
+2024-02-23 14:55:21 [info]:     Font-Commerce Server Events Worker
+2024-02-23 14:55:21 [info]:     Please wait booting...
+2024-02-23 14:55:24 [info]:     Worker started
+```
+
+And now let's head to the faq question page and vote.
+
+You will see new output to the worker:
+
+```
+[ConsoleIntegration]: (1708696560088-0) {"event_type":"QuestionVotedEvent","created_at":"2024-02-23T13:56:00.080Z","payload":"{\"questionSlug\":\"probleme-colis\",\"upVote\":true}","metadata":"{\"shopId\":\"default\"}"}
+```
+
+And that's it! You now know how to create your own event emitter and send
+server-side events.
```

Comment:
**issue:** This guide is about creating and sending server events, not creating emitters. To be clear, "creating a server emitter" is (and should be) only done one time [here](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/events/domain/UserJourney.ts#L9) in Front-Commerce core. It is not the responsibility of the integrators to do so.

```suggestion
And that's it! You now know how to create and emit your own server events.
```

Reply:
Fixed with your suggestion